src/course-unit/sidebar/utils.jsx

```diff
@@ -0,0 +1,100 @@
+import {
+  CheckCircle as CheckCircleIcon,
+  CheckCircleOutline as CheckCircleOutlineIcon,
+  InfoOutline as InfoOutlineIcon,
+} from '@edx/paragon/icons';
+
+import { COLORS, UNIT_VISIBILITY_STATES } from '../constants';
+import messages from './messages';
+
+/**
+ * Get information about the publishing status.
+ * @param {Object} intl - The internationalization object.
+ * @param {boolean} hasChanges - Indicates if there are unpublished changes.
+ * @param {string} editedBy - The user who edited the content.
+ * @param {string} editedOn - The timestamp when the content was edited.
+ * @param {string} publishedBy - The user who last published the content.
+ * @param {string} publishedOn - The timestamp when the content was last published.
+ * @returns {string} Publish information based on the provided parameters.
+ */
+export const getPublishInfo = (intl, hasChanges, editedBy, editedOn, publishedBy, publishedOn) => {
+  if (hasChanges && editedOn && editedBy) {
+    return intl.formatMessage(messages.publishInfoDraftSaved, { editedOn, editedBy });
+  } if (publishedOn && publishedBy) {
+    return intl.formatMessage(messages.publishLastPublished, { publishedOn, publishedBy });
+  }
+
+  return intl.formatMessage(messages.publishInfoPreviouslyPublished);
+};
+
+/**
+ * Get information about the release status.
+ * @param {Object} intl - The internationalization object.
+ * @param {string} releaseDate - The release date of the content.
+ * @param {string} releaseDateFrom - The section name associated with the release date.
+ * @returns {string|ReactElement} Release information based on the provided parameters.
+ */
+export const getReleaseInfo = (intl, releaseDate, releaseDateFrom) => {
+  if (releaseDate) {
+    return (
+      <span className="course-unit-sidebar-date-and-with">
+        <h6 className="course-unit-sidebar-date-timestamp m-0 d-inline">
+          {releaseDate}&nbsp;
+        </h6>
+        {intl.formatMessage(messages.releaseInfoWithSection, { sectionName: releaseDateFrom })}
+      </span>
+    );
+  }
+
+  return intl.formatMessage(messages.releaseInfoUnscheduled);
+};
```

Comment:
```suggestion
export const getReleaseInfo = (intl, releaseDate, releaseDateFrom) => {
  if (releaseDate) {
    return {
      isScheduled: true,
      releaseDate,
      releaseDateFrom,
      sectionNameMessage: intl.formatMessage(messages.releaseInfoWithSection, { sectionName: releaseDateFrom }),
    };
  }

  return {
    isScheduled: false,
    message: intl.formatMessage(messages.releaseInfoUnscheduled),
  };
};
```

We don't use `JSX` in utils, please change the usage of this function

Reply:
```
import React from 'react';
import { getReleaseInfo } from './utils';

const ReleaseInfoComponent = ({ intl, releaseDate, releaseDateFrom }) => {
  const releaseInfo = getReleaseInfo(intl, releaseDate, releaseDateFrom);

  if (releaseInfo.isScheduled) {
    return (
      <span className="course-unit-sidebar-date-and-with">
        <h6 className="course-unit-sidebar-date-timestamp m-0 d-inline">
          {releaseInfo.releaseDate}&nbsp;
        </h6>
        {releaseInfo.sectionNameMessage}
      </span>
    );
  }

  return <>{releaseInfo.message}</>;
};
```

Reply:
Created